src/mapgen_functions.cpp

```diff
@@ -3494,182 +3494,6 @@ void mapgen_police(map *m, oter_id terrain_type, mapgendata dat, int, float dens
 }
 
 
-void mapgen_bank(map *m, oter_id terrain_type, mapgendata dat, int, float)
-{
-    dat.fill_groundcover();
-    // Basic floorplan
-    square(m, t_floor, 1,  1, 22, 22);
-    line(m, t_wall,  1,  1, 22,  1);
-    line(m, t_wall,  2,  5,  5,  5);
-    line(m, t_wall, 16,  5, 19,  5);
-    line(m, t_wall,  2,  9, 19,  9);
-    line(m, t_wall, 12, 12, 21, 12);
-    line(m, t_wall,  2, 14,  6, 14);
-    line(m, t_wall, 13, 16, 21, 16);
-    line(m, t_wall,  1, 22, 22, 22);
-    line(m, t_wall,  1,  2,  1, 21);
-    line(m, t_wall, 22,  2, 22, 21);
-    line(m, t_wall,  7, 10,  7, 21);
-    line(m, t_wall, 12, 12, 12, 21);
-    line(m, t_wall, 19, 13, 19, 15);
-    line(m, t_wall, 16,  6, 16,  8);
-    line(m, t_wall, 19,  6, 19,  8);
-    line(m, t_wall_metal,  2, 15,  6, 15);
-    line(m, t_wall_metal,  2, 21,  6, 21);
-    line(m, t_wall_metal,  2, 16,  2, 20);
-    line(m, t_wall_metal,  6, 16,  6, 20);
-    //Fixed doors
-    line(m, t_door_glass_c, 9, 1, 10, 1);
-    m->ter_set( 19,  6, t_door_c);
-    m->ter_set( 12, 17, t_door_c);
-    m->ter_set( 17, 12, t_door_c);
-    square(m, t_door_metal_locked, 6, 19, 7, 20);
-    if (one_in(3)) {
-        line(m, t_door_locked_interior, 20, 9, 21, 9);
-    } else {
-        line(m, t_door_c, 20, 9, 21, 9);
-    }
-    if (one_in(4)) {
-        m->ter_set( 7, 10, t_door_locked_interior);
-    } else {
-        m->ter_set( 7, 10, t_door_c);
-    }
-    //Lobby
-    line(m, t_atm,  2,  2,  2,  3);
-    if (one_in(2)) {
-        line_furn(m, f_chair, 13, 2, 15, 2);
-        m->furn_set(14, 2, f_table);
-    }
-    if (one_in(2)) {
-        line_furn(m, f_chair, 17, 2, 19, 2);
-        m->furn_set(18, 2, f_table);
-    }
-    if (one_in(2)) {
-        m->furn_set(21, 2, f_indoor_plant);
-    }
-    //Windows or glass wall front?
-    int tmp = 0;
```

Comment:
Is there any way to retain the old random terrain generation?

Reply:
Multiple definitions - one per variant.

Reply:
Indoor plants are random - they can be spawned with 50% probability. I've also restored random locked doors.

Other randomness is not possible now in JSON - only in LUA.